test/UID2.Client.Test/SharingClientTests.cs

```diff
@@ -0,0 +1,420 @@
+        [InlineData(IdentityScope.EUID, TokenVersion.V2)]
+        [InlineData(IdentityScope.UID2, TokenVersion.V3)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V3)]
+        [InlineData(IdentityScope.UID2, TokenVersion.V4)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V4)]
+        private void TokenGeneratedInTheFutureLegacyClient(IdentityScope identityScope, TokenVersion tokenVersion)
+        {   //similar to BidstreamClientTests.TokenGeneratedInTheFutureLegacyClient, but uses KeySharingResponse and Decrypt without domain parameter
+            UID2Client client = new("endpoint", "authkey", CLIENT_SECRET, identityScope);
+
+            var refreshResult = client.RefreshJson(KeySharingResponse(new[] { MASTER_KEY, SITE_KEY }, identityScope));
+            Assert.True(refreshResult.Success);
+
+            var advertisingToken = AdvertisingTokenBuilder.Builder().WithVersion(tokenVersion).WithScope(identityScope).WithGenerated(DateTime.UtcNow.AddMinutes(31)).Build();
+
+            var res = client.Decrypt(advertisingToken);
+            BidstreamClientTests.AssertSuccess(res, tokenVersion);
+        }
+
+        [Theory]
+        [InlineData(IdentityScope.UID2, TokenVersion.V2)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V2)]
+        [InlineData(IdentityScope.UID2, TokenVersion.V3)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V3)]
+        [InlineData(IdentityScope.UID2, TokenVersion.V4)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V4)]
+        private void TokenLifetimeTooLongLegacyClient(IdentityScope identityScope, TokenVersion tokenVersion)
+        {    //similar to BidstreamClientTests.TokenLifetimeTooLongLegacyClient, but uses KeySharingResponse and Decrypt without domain parameter
+            UID2Client client = new("endpoint", "authkey", CLIENT_SECRET, identityScope);
+
+            var refreshResult = client.RefreshJson(KeySharingResponse(new[] { MASTER_KEY, SITE_KEY }, identityScope));
+            Assert.True(refreshResult.Success);
+
+            var advertisingToken = AdvertisingTokenBuilder.Builder().WithVersion(tokenVersion).WithScope(identityScope).WithExpiry(DateTime.UtcNow.AddDays(3).AddMinutes(1)).Build();
+
+            var res = client.Decrypt(advertisingToken);
+            BidstreamClientTests.AssertSuccess(res, tokenVersion);
+        }
+
+
+        // tests below taken from EncryptionTestsV4.cs under "//  Sharing tests" comment and modified to use SharingClient and the new JSON /key/sharing response
+        private SharingClient SharingSetupAndEncrypt(out string advertisingToken)
+        {
+            var json = KeySetToJsonForSharing(new [] {MASTER_KEY, SITE_KEY});
+            var refreshResult = _client.RefreshJson(json);
+            Assert.True(refreshResult.Success);
+
+            advertisingToken = SharingEncrypt(_client);
+
+            return _client;
+        }
+
+        private string SharingEncrypt(SharingClient client)
```

Comment:
could you combine this with `EuidClientProducesEuidToken` below by using [Theory] and also calling `ValidateAdvertisingToken` to validate token values?